services/QuillLMS/client/app/bundles/Connect/components/questions/responseComponent.jsx

```diff
@@ -485,7 +488,17 @@ class ResponseComponent extends React.Component {
   };
 
   renderResponses = () => {
-    if (this.state.viewingResponses) {
+    const { isLoadingResponses, viewingResponses } = this.state
+
+    if (isLoadingResponses && viewingResponses) {
+      return (
+        <div className="loading-spinner-container">
+          <Spinner />
+        </div>
+      );
+    }
+
+    if (viewingResponses) {
```

Comment:
Not a huge deal, but you could re-arrange your if conditionals with an early return:
```javascript
if (!viewingResponses) return
if (isLoadingResponses) { return ( ... spinner stuff ) }
return <ResponseList ...
```